meta-balena-common/recipes-bsp/grub/grub-efi_%.bbappend

```diff
@@ -1,9 +1,15 @@
-# We don't want grub modules in our sysroot, remove the entire prefix after do_deploy
-# This removes them for x86-64
-do_deploy_append_class-target_x86-64() {
-    cp -r ${D}${libdir}/grub/ ${DEPLOYDIR}/
+# We don't want grub modules in our sysroot
+do_install_append_class-target() {
     rm -rf ${D}${prefix}
 }
 
-# also remove the entry from FILES
-FILES_${PN}_remove_class-target_x86-64 = "${libdir}/grub/${GRUB_TARGET}-efi"
+# Reinstall modules in DEPLOYDIR to be installed in resin-image
+do_deploy_append_class-target() {
+    oe_runmake 'libdir=${DEPLOYDIR}' -C grub-core install
```

Comment:
@jakogut is this installing all grub modules? Would it be better to copy just the modules we need? Or do we only build what we need?

Reply:
Good question. It seems we've [always installed](https://github.com/balena-os/balena-intel/blob/3746feceb5b42cee05fca95c66b81b59b34e68bd/layers/meta-balena-genericx86/recipes-bsp/grub/grub-efi_%25.bbappend#L6) all of the release modules, and this build target is no exception.

However, there is a variable in the grub-efi [recipe](https://github.com/balena-os/balena-intel/blob/3746feceb5b42cee05fca95c66b81b59b34e68bd/layers/meta-balena-genericx86/recipes-bsp/grub/grub-efi_2.04.bb#L49), `GRUB_BUILDIN`, that defines a list of modules to build into the grub image. If we were to amend that with the required modules, we shouldn't need to install external modules anymore. Maybe we could default to the built in modules, and add a variable to optionally install external modules?

Reply:
@jakogut built-in modules for an embedded device seems a good option to me. Why not take a look see how that works?

Reply:
Will do.